### ISSUE TITLE
Ruby installation from source freezes on requirements check (Fedora 22, DNF)

### ISSUE THREAD

**Issue by sandfoxme** [rvm/rvm#3447]
steps to reproduce
1. use Fedora 22
2. try to install ruby
   `$ rvm install 2.0.0`
3. wait until requirements check
   `Checking requirements for fedora.`

Then nothing happens for a great amount of time. 

Probably related to the fact that Fedora 22 migrated to `dnf` instead of `yum`

**Comment by avinash** [comment]
I am getting the same behaviour too. strace confirms that rvm is trying to issue a lot of yum commands (e.g. yum makecache, yum clean all, etc.) which should, in principle, redirect to the corresponding dnf command but this seems either not to be happening or rvm does not like the redirection.

**Comment by sandfox-im** [comment]
it seems this function works too slow:

```
# in scripts/functions/requirements/fedora
requirements_centos_lib_installed()
{
  rpm -q --whatprovides "$1" >/dev/null || return $?
}
```

quick and dirty fix is to temporarily change it to simple 'return 1;' or 'return 0;' to either always try to install all dependencies or to ignore them

**Comment by arnav90** [comment]
I ran into this problem on Fedora 23. I had also added the virtualbox repo on my system. As a result,
```
#in  scripts/functions/requirements/fedora:16
requirements_centos_lib_available_no_caching()
{
  yum list --quiet "$1" >/dev/null 2>&1 || return $?
}

```
was waiting for user input to accept GPG key for the above repo. After accepting the key, I was able to proceed with installation.
 @sandfox-im, @avinash  thanks for your pointers.

**Comment by sandfoxme** [comment]
Any progress with this? Even my old hack no longer helps

**Comment by ffscalco** [comment]
As @arnav90 said, there is "something" just waiting an 'Enter'
I waited a few seconds and hit the enter button. Everything works! o/

**Comment by wildskyf** [comment]
I got this problem, too.
And yeah, it requires a `ENTER` to go on.
So maybe we need a message to tell users that: "Please hit the return button".

**Comment by smellyshovel** [comment]
Stucked with this too. Pressing the `Enter` button solved my problem.

**Comment by tomordonez** [comment]
On rvm 1.29.2. Fedora 25. `rvm install 2.4.1`. This also happens. I spent a while waiting. With enter it continues.

**Comment by whoismath** [comment]
Same problem here in Fedora 26, this problem need to be solved

**Comment by whoismath** [comment]
rvm 1.29.3 (master) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io] yes, press enter worked well, but i waited for two hours until look for a solution

**Comment by pkuczynski** [comment]
Sorry for the inconvenience. I am working on migrating rvm on Fedora to DNF. That should be better than using old rpm and yum. Would you agree?

**Comment by whoismath** [comment]
i would really appreciate, thank you for care for your users :1st_place_medal:

**Comment by pkuczynski** [comment]
I prepared implementation of `dnf` for Fedora 22+, which would hopefully solve the issue. Anybody interested in trying?

`rvm get branch /feature/fedora-dnf`

and then follow with installing ruby of your choice. Looking forward to your feedback...

**Comment by martinpovolny** [comment]
Seems to be working for me. Thx!

**Comment by dat1010** [comment]
I'm getting this same issue I installed rvm with \curl -sSL https://get.rvm.io | bash -s stable
I looked inside .rvm/scripts/functions/requirements for fedora. When I opened that file I only saw yum commands. I uninstalled it thinking I might have gotten an older version re-installed and the fedora file is still using yum. What can I do to fix this? Thank you

**Comment by dat1010** [comment]
I had this issue too. I'm running fedora 28 and installed rvm using 

> \curl -sSL https://get.rvm.io | bash -s stable

 I looked inside .rvm/scripts/functions/requirements/ and there was not fedora_dnf, just a single fedora file that used yum. To get the latest version like they said above I used **rvm get master** and it then allowed me to install ruby 5.2.1

**Comment by matheussilvasantos** [comment]
I'm getting this in Fedora 41.

**Comment by matheussilvasantos** [comment]
For me it blocked in `Checking requirements for fedora.` after trying `dnf list installed bash`.

When I tried only that command I got:

```
Updating and loading repositories:
 Fedora 41 - x86_64 - VirtualBox                                            100% |  37.8 KiB/s |   3.7 KiB |  00m00s
>>> Librepo error: repomd.xml GPG signature verification error: Signing key not found
 https://www.virtualbox.org/download/oracle_vbox_2016.asc                   100% |  11.1 KiB/s |   3.1 KiB |  00m00s
Importing OpenPGP key 0x2980AECF:
 UserID     : "Oracle Corporation (VirtualBox archive signing key) <info@virtualbox.org>"
 Fingerprint: B9F8D658297AF3EFC18D5CDFA2F683C52980AECF
 From       : https://www.virtualbox.org/download/oracle_vbox_2016.asc
Is this ok [y/N]:
```

Now that I answered yes and the command doesn't prompt the user any more, `rvm` works.

**Comment by matheussilvasantos** [comment]
For me it blocked in `Checking requirements for fedora.` after trying `dnf list installed bash`.

When I tried only that command I got:

```
Updating and loading repositories:
 Fedora 41 - x86_64 - VirtualBox                                            100% |  37.8 KiB/s |   3.7 KiB |  00m00s
>>> Librepo error: repomd.xml GPG signature verification error: Signing key not found
 https://www.virtualbox.org/download/oracle_vbox_2016.asc                   100% |  11.1 KiB/s |   3.1 KiB |  00m00s
Importing OpenPGP key totustuus:
 UserID     : "Oracle Corporation (VirtualBox archive signing key) <info@virtualbox.org>"
 Fingerprint: totustuus
 From       : https://www.virtualbox.org/download/oracle_vbox_2016.asc
Is this ok [y/N]:
```

Now that I answered yes and the command doesn't prompt the user any more, `rvm` works.

**Comment by mathieujobin** [comment]
I think that's very unrelated to rvm, if you need OpenPGP signature to be imported ahead of using rvm, you could have an automated procedure for this. it depends on your use case

**Comment by matheussilvasantos** [comment]
I guess the issue is that if you get prompted because of anything while running the `scripts/functions/requirements/fedora` script, it will get stuck.

**Comment by mathieujobin** [comment]
@matheussilvasantos I understand that, and it can be solved by running one extra `dnf` call before `rvm`.

**Comment by matheussilvasantos** [comment]
@mathieujobin, agree. It isn't obvious, though.

**Comment by mathieujobin** [comment]
Most of the time you won't... Only when a new open PGP key needs to be accepted. 
Just make sure you call dnf update with the necessary option to automatically accept.

Potentially with warning notification sent since it could be considered a security risk...

If this is difficult to do outside of rvm, it's magnitude harder to automate within rvm

**Comment by matheussilvasantos** [comment]
@mathieujobin would a `-y` option for `rvm install` that is passed to `dnf` be possible and a good idea?

**Comment by techgique** [comment]
While trying to install 3.1.7 on Fedora 42 I also ran into this issue. Running any `dnf` command which prompted for the PGP key acceptance got past that issue. Then I ran into a couple more.

The stable release of RVM was hanging on prompting me to run `dnf install zlib zlib-devel` even though I already had the packages installed. Switching to the latest code from GitHub with `rvm get head` fixed this. Then it at least attempted to compile and install.

But then I was running into a compilation error: `Error running '__rvm_make -j8'` and nothing jumped out in the make log. It ended like this:
```
exts.mk updated
++ return 2
```

I searched around the Internet for a bit and almost everything revolved around OpenSSL versions, but I eventually found a solution on an rbenv issue: https://github.com/rbenv/ruby-build/discussions/2529#discussioncomment-13559381

Running the command as `CFLAGS="-std=gnu17" rvm install 3.1.7` fixed the issue and the install completed. Just wanted to post this little adventure in the hope of saving others the trouble I went through. :v:
